### DISCUSSION TITLE
Shared Data, updates trying to change ownership of shared folder - not allowed

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#760] [Fixed Issues] [answered]
@rern 

Just applied update 20220116 to my Shared Data installation. It seems to have a problem updating the ownership of the all the Shared Data items at the end of the update - as shown in the attached image of part of the update logs.  I am using an NFS share for the shareddata folder as below:

rAudio-1:/srv/http/shareddata  #  ls -lrt
total 28
drwxr-sr-x+ 2 1007 100 4096 Jan  8 15:15 audiocd
drwxr-sr-x+ 2 1007 100 4096 Jan  8 15:15 bookmarks
drwxr-sr-x+ 2 1007 100 4096 Jan  8 15:15 lyrics
drwxr-sr-x+ 2 1007 100 4096 Jan  8 15:15 playlists
drwxr-sr-x+ 2 1007 100 4096 Jan  9 08:45 webradiosimg
drwxr-sr-x+ 2 1007 100 4096 Jan  9 08:45 mpd
drwxr-sr-x+ 2 1007 100 4096 Jan  9 08:45 webradios

The high level folder was created in OpenmediaVault v5.x and rAudio-1 had no problem writing the original files to this location and using them since. Why then does the ownership need to be changed with this update ? 

Thanks 

Chris
[Capture160122.zip](https://github.com/rern/rAudio-1/files/7876658/Capture160122.zip)

**Comment by ceejayemm** [comment]
@rern 

I am not sure that the Shared Data option is working as intended.  This morning I connected to my Office installation and could not play an album.  On checking my Library / NAS I found I only had 5 albums listed. I rebuilt the entire Library and I now have 321 albums listed on the Office instance.  

Switching to my Loft installation I checked the Library / NAS information (after the Library rebuild above) and I have the same 5 albums listed as originally found on the Office instance - not the 321 I would have expected.  I have refreshed the browser (Chrome - several times), tried a different browser on another PC (Brave - Linux installation) still the same and rebooted the RPI several times. I still have the same 5 albums listed. Both RPi's are setup to use the same Shared Data area and also the same music library - both attached to the RPI via NFS. 

The only issue I know of with these machines is the attempted change of ownership issue noted above when applying an update. I can see that the access date/time of the files stored on the Shared Data area are changing when the Library update is performed so, despite the problems when the update is applied, the Shared Data does seem to be getting updated when a Library change is detected. 

As I wrote the above an idea occurred to me.  As noted above the Office instance, after a Library update, was showing 321 albums in the Library / NAS area. On the Loft instance, which I have been using to test the issues in #767, only 5 albums were being shown. It was this machine where I had applied and tested the latest updates up to and including 20220117.  I had not applied this update to the Office instance, which I have now done.  Lo and behold the Library / NAS area on the Office instance now also shows only 5 albums. AND these 5 albums are the only ones I possess which are made up of .mp3 files - all the rest of my albums are made up of .m4a files.  I have checked, on both instances the Player / Options / FFmpeg option IS set to ON and has been all the time during this and previous testing.  

So, is the Library / NAS update not being performed because of the errors during the change of ownership step in the update process ?  If I run a Full Library Update the Library / NAS information IS being updated (on the machine where it is run (local files ?)) but not on the Shared Data area - why is that ?

Chris

**Comment by ceejayemm** [comment]
@rern 

I have changed the title of this issue to more clearly identify the problem.

As noted above the Shared Data files are being created on the (NFS, in my case) share correctly on first use but all subsequent updates ( up to and including 20220122) try to change the ownership of the files at the end of the update - which fails because the files are owned by the NFS server not the RA client (details shown in notes above). This fails for ALL files in the Shared Data folders / files not just those show nas examples above. The Library data held on the Shared Data is then updated with ONLY the albums /files stored as .MP3 files. This makes the Library unusable for any other files held on the Music share as their location is 'not known' to RA.

Why do the updates try to change the ownership of the Shared Data files / folders when they effectively no longer belong to the RA installation but are managed by the NFS server ?

I have been experimenting  with a MPD Satellite setup for the music library which works well in this kind of shared environment but does not cater for the 'extra' RA items (counts, genres, artists, etc etc) so the Shared Data share is still need for them.

Chris

**Reply by rern** [reply]
Hi Chris,

Please try the UPDATE branch again.

**Comment by ceejayemm** [comment]
@rern 

I have applied the latest UPDATE branch but still have the same problem.  The update tries to 'chown' the Shared Data area and produces 'Operation not permitted' on all files in this area.

Chris

**Reply by rern** [reply]
Hi Chris,

Try creating a new share with max permissions:
- Directory: `777`
- NFS Share: `rw`

**Reply by ceejayemm** [reply]
@rern 

I have done as you requested but it makes no difference, the final part of the update (using the UPDATE branch) fails with the same error.

I don't believe its possible to chown/chmod an NFS share from the Client as you are trying  to do. Such changes can only be made on the NFS server. If I can write the Shared Data files in the first place and I can access then afterwards why do you need to check/change the ownership/permissions with each update ? I know making the share 'open' (D: 777 / rw) as you suggest should make it easy for you work with but it does nothing for any form of peace of mind / security :-)

Chris

**Reply by rern** [reply]
Hi Chris,

Ownership and permissions are for local directories only. The update script was revised to be more specific for the directories.
Please try again.

**Reply by ceejayemm** [reply]
@rern 

I have applied the latest UPDATE branch and it now seems to complete the update without any issues. 

Many thanks once again for sorting this problem.

Chris

**Comment by ceejayemm** [comment, marked as answer]
Answer as suggested by @rern
